torch/csrc/jit/passes/onnx.cpp

```diff
@@ -293,14 +293,18 @@ void NodeToONNX(
         if (old->hasDebugName() && !exist_in_env) {
           auto old_name = outputs[i]->debugName();
           auto new_name = old->debugNameBase();
-          auto debug_names = new_block->owningGraph()->debugNames();
+          const auto& debug_names = new_block->owningGraph()->debugNames();
           auto exist_name = debug_names.find(new_name);
+          Value* found_value;
+          bool exists = exist_name != debug_names.end();
+          if (exists) {
+            found_value = exist_name->second;
+          }
+          // Note: calls to setDebugName will change debug_names and corrupt
+          // exist_name
```

Comment:
The original comments seem to be clearer, (even though I don't understand the behavior completely). Do you have an idea of what happens here; could you potentially improve the note?

Reply:
This is really subtle, so I have taken a stab at making this even more clear. I'll let the code change speak for itself, so let me know if it's clear now.